src/pages/Admin/Room/Edit/RoomEditModal.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useEffect, useState, useCallback, useMemo } from "react";
+import { Form, Modal, Button } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import ImageInput from "components/atoms/ImageInput";
+
+import { RoomTemplate, ROOM_TEMPLATES } from "settings";
+import { RoomEditModalProps } from "./RoomEditModal.types";
+
+import "./RoomEditModal.scss";
+
+export const RoomEditModal: React.FC<RoomEditModalProps> = ({
+  isVisible,
+  onClickOutsideHandler,
+  room,
+  submitHandler,
+  deleteHandler,
+}) => {
+  const [selectedTemplate, setSelectedTemplate] = useState<string>(
+    room.template!
+  );
+
+  const [roomTemplate, setRoomTemplate] = useState<RoomTemplate | null>(null);
+
+  useEffect(() => {
+    const template = ROOM_TEMPLATES.find(
+      (i) => i.template === selectedTemplate
+    );
+    setRoomTemplate(template!);
+  }, [selectedTemplate]);
```

Comment:
Removing this typescript 'force resolution' hack, as it's not safe. This may cause other type issues to show up, my guess is that it will be potentially undefined, since that's what this function returns when it doesn't find anything:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Which will probably cause type issues with the `| null` above. I expect that can probably just become `| undefined` + fix up the relevant areas that consume it.

```suggestion
  useEffect(() => {
    const template = ROOM_TEMPLATES.find(
      (i) => i.template === selectedTemplate
    );
    setRoomTemplate(template);
  }, [selectedTemplate]);
```